libs/knowledge-store/ragstack_knowledge_store/directed_edge_extractor.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import annotations
+
+from typing import Any, Dict, Iterable, Set
+
+from ragstack_knowledge_store.edge_extractor import EdgeExtractor
+from ragstack_knowledge_store.knowledge_store import CONTENT_ID, KnowledgeStore
+
+
+class DirectedEdgeExtractor(EdgeExtractor):
+    """Extract edges between uses (of a URL or other ID) and definitions (of a URL or other ID).
+
+    While `UndirectedEdgeExtractor` links nodes in both directions if they share
+    a keyword, this only creates links from nodes with a "source" to nodes with
+    a matching "target". For example, uses may be the `href` of `a` tags in the
+    chunk and definitions may be the URLs that the chunk is accessible at.
+
+    This may also be used for other forms of references, such as Wikipedia article IDs, etc.
+    """
+
+    def __init__(self, sources_field: str, targets_field: str, kind: str) -> None:
+        """Create a new DirectedEdgeExtractor.
```

Comment:
I propose to merge the init docstring with the class docstring and only put it on init.
IDEs and online doc builders will use the init docstring for the class if the class doesn't have one.